src/wordy/ast/BinaryExpressionNode.java

```diff
@@ -59,4 +62,57 @@ public String toString() {
     protected String describeAttributes() {
         return "(operator=" + operator + ')';
     }
+
+    @Override
+    protected double doEvaluate(EvaluationContext context) {
+        double left = lhs.evaluate(context);
+        double right = rhs.evaluate(context);
+        
+        switch(operator) {
+            case ADDITION:
+                return left + right;
+            case SUBTRACTION:
+                return left - right;
+            case MULTIPLICATION:
+                return left * right;
+            case DIVISION:
+                return left / right;
+            case EXPONENTIATION:
+                return Math.pow(left, right);
+        } 
+
+        return 0.0;
+        
+    }
+
+    @Override
+    public void compile(PrintWriter out) {
```

Comment:
This method has the same issue as the previous one. My code and Paul's code do essentially the same thing, but mine doesn't protect against invalid operators